internal/daemon/config.go

```diff
@@ -28,65 +34,70 @@ func (c *ConfigFile) SetDefaults() {
 	}
 }
 
-// Assert interface compliance.
-var _ defaults.Setter = (*ConfigFile)(nil)
-
-// config holds the configuration state as a singleton. It is used from LoadConfig and Config
-var config *ConfigFile
-
-// LoadConfig loads the daemon config from given path. Call it only once when starting the daemon.
-func LoadConfig(path string) error {
-	if config != nil {
-		return errors.New("config already set")
+// Validate implements the config.Validator interface.
+// Validates the entire daemon configuration on daemon startup.
+func (c *ConfigFile) Validate() error {
+	if err := c.Database.Validate(); err != nil {
+		return err
 	}
-
-	cfg, err := fromFile(path)
-	if err != nil {
+	if err := c.Logging.Validate(); err != nil {
 		return err
 	}
 
-	config = cfg
-
 	return nil
 }
 
-// Config returns the config that was loaded while starting the daemon
-func Config() *ConfigFile {
-	return config
+// Assert interface compliance.
+var (
+	_ defaults.Setter  = (*ConfigFile)(nil)
+	_ config.Validator = (*ConfigFile)(nil)
+)
+
+// Flags defines the CLI flags supported by Icinga Notifications.
+type Flags struct {
+	// Version decides whether to just print the version and exit.
+	Version bool `long:"version" description:"print version and exit"`
+	// Config is the path to the config file
+	Config string `short:"c" long:"config" description:"path to config file"`
 }
 
-func fromFile(path string) (*ConfigFile, error) {
-	f, err := os.Open(path)
-	if err != nil {
-		return nil, err
+// daemonConfig holds the configuration state as a singleton.
+// It is initialised by the ParseFlagsAndConfig func and exposed through the Config function.
+var daemonConfig *ConfigFile
+
+// Config returns the config that was loaded while starting the daemon.
+// Panics when ParseFlagsAndConfig was not called earlier.
+func Config() *ConfigFile {
+	if daemonConfig == nil {
+		panic("ERROR: daemon.config not initialized")
```

Comment:
This now references an outdated variable name. Also, the variable name is internal to the package, so I'd give the panic message the same treatment as the comment and say that `ParseFlagsAndConfig()` wasn't called.